src/mqt/predictor/rl/predictorenv.py

```diff
@@ -284,6 +286,9 @@ def apply_action(self, action_index: int) -> QuantumCircuit | None:
 
             elif action["origin"] == "tket":
                 try:
+                    self.state = self.state.decompose(
+                        gates_to_decompose=["ecr", "rzx"], reps=2
+                    )  # TKET cannot handle ecr gates and two decompositions are necessary because one leads to rzx gates that cannot be handled as well
```

Comment:
I have a hard time believing this comment as tket does support the ECR gate.
https://github.com/CQCL/pytket-qiskit/blob/2b358b5641687e96dfee4f163a2ce012ceb050fe/pytket/extensions/qiskit/qiskit_convert.py#L128-L148

It does not support the `RZX` gate though. (this would probably be a nice addition to tket and pytket-qiskit)

Reply:
This does not matter anymore due to the changes from above.